WalletWasabi/Wallets/Wallet.cs

```diff
@@ -63,6 +63,7 @@ public class Wallet : BackgroundService, IWallet
 		WalletFilterProcessor = new WalletFilterProcessor(KeyManager, BitcoinStore, TransactionProcessor, BlockProvider);
 		BatchedPayments = new PaymentBatch();
 		OutputProvider = new PaymentAwareOutputProvider(DestinationProvider, BatchedPayments);
+		Id = ++WalletCount;
```

Comment:
Concurrency can be a problem here

Reply:
We can't create wallets from different multiple threads. Can we?

Reply:
At the moment we don't do it, but who knows in the future. I just want to be absolutely sure ID is always unique. Imagine having two wallets with the same ID, that could be a disaster.

How about:
```c#
Id = Interlocked.Increment(ref WalletCount);
```